src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,99 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stderr or any other stream.
+ *
+ * @internal
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Logger extends AbstractLogger
+{
+    private static $levels = array(
+        LogLevel::DEBUG => 0,
+        LogLevel::INFO => 1,
+        LogLevel::NOTICE => 2,
+        LogLevel::WARNING => 3,
+        LogLevel::ERROR => 4,
+        LogLevel::CRITICAL => 5,
+        LogLevel::ALERT => 6,
+        LogLevel::EMERGENCY => 7,
+    );
+
+    private $minLevelIndex;
+    private $formatter;
+    private $handle;
+
+    public function __construct($minLevel = LogLevel::WARNING, $output = 'php://stderr', callable $formatter = null)
+    {
+        if (!isset(self::$levels[$minLevel])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $minLevel));
+        }
+
+        $this->minLevelIndex = self::$levels[$minLevel];
+        $this->formatter = $formatter ?: array($this, 'format');
+        if (false === $this->handle = @fopen($output, 'a')) {
+            throw new InvalidArgumentException(sprintf('Unable to open "%s".', $output));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset(self::$levels[$level])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if (self::$levels[$level] < $this->minLevelIndex) {
+            return;
+        }
+
+        $formatter = $this->formatter;
+        fwrite($this->handle, $formatter($level, $message, $context));
+    }
+
+    /**
+     * @param string $level
+     * @param string $message
+     *
+     * @return string
+     */
+    private function format($level, $message, array $context)
+    {
+        if (false !== strpos($message, '{')) {
+            $replacements = array();
+            foreach ($context as $key => $val) {
+                if (null === $val || is_scalar($val) || (\is_object($val) && method_exists($val, '__toString'))) {
+                    $replacements["{{$key}}"] = $val;
+                } elseif ($val instanceof \DateTimeInterface) {
+                    $replacements["{{$key}}"] = $val->format(\DateTime::RFC3339);
+                } elseif (\is_object($val)) {
+                    $replacements["{{$key}}"] = '[object '.\get_class($val).']';
+                } else {
+                    $replacements["{{$key}}"] = '['.\gettype($val).']';
+                }
+            }
+
+            $message = strtr($message, $replacements);
+        }
+
+        return sprintf('[%s] %s', $level, $message).\PHP_EOL;
```

Comment:
Should we add the date in the default formatter (like Go does)?

Reply:
I guess we should. Or could the infra's log collector be responsible for adding the date? Is that common practice?

Reply:
Docker doesn't do it by default at least.

Reply:
would be useful yes but default date format & precision should be decided in this case.

Reply:
I would suggest RFC 3339 as this is a standard we use everywhere else.

Reply:
The only issue with `RFC 3339` is sometime you want a precision under the second but ok should be addressed by folks with the callable formatter for example.

Reply:
RFC 3339 is the way to go to me also, and a custom formatter/monolog would overdo it for anyone needing more/another format.